src/snowflake/snowpark/stored_procedure_profiler.py

```diff
@@ -72,6 +75,10 @@ def set_active_profiler(
             (case-sensitive). Active profiler is 'LINE' by default.
 
         """
+        if not self._has_target_stage:
+            raise ValueError(
+                "Stored procedure profiler does not have a valid target stage, please provide one in profiler.set_target_stage()."
+            )
```

Comment:
Can we use some default temporary stage i.e. from [Session.get_session_stage](https://docs.snowflake.com/en/developer-guide/snowpark/reference/python/latest/snowpark/api/snowflake.snowpark.Session.get_session_stage#snowflake.snowpark.Session.get_session_stage) here?

Reply:
yes, I think we can do that, then we actually don't need to raise this error because there is always a targeted stage then.

Reply:
+1